Source/EasyNetQ/DI/IConsumeScopeProvider.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+
+namespace EasyNetQ.DI
+{
+    /// <summary>
+    /// Provides dependency resolution scope for <see cref="RabbitAdvancedBus.Consume(Action{IConsumeConfiguration})"/>
+    /// </summary>
+    public interface IConsumeScopeProvider
+    {
+        /// <summary>
+        /// Creates scope
+        /// </summary>
+        IDisposable CreateScope();
```

Comment:
I would expect this to return a scope, not IDisposable (I don't think IDisposable should be used as return types in public interfaces)

Reply:
`IServiceResolverScope` ? Well, this **can** be done, but firstly, it does not need (at least for now), because the scope methods are not used, it is only created. Secondly, then it will require to implement `IServiceResolverScope/IServiceResolver` from external implementations. For my implementation now it will be just a burden. I would prefer to return `IDisposable` as the smallest sufficient requirement.

Reply:
If @Pliner @micdenny agree sure, but to me this doesn't look OK, it feels like an unintuitive API, a leaky abstraction, and there's a high chance we have to break it later to make the return type more specific. If you have a method that is named CreateScope, it should return a scope, regardless of your current requirements. But that's just my 2cts.

Reply:
Makes sense (:

Reply:
done